File: src/tn/esprit/YoTalent/entities/Evenement.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package tn.esprit.YoTalent.entities;

import java.util.Objects;

/**
 *
 * @author USER
 */
public class Evenement {
      private int  idEv;
    private String nomEv,localisation,dateDEv,dateFEv;
    
    

    public Evenement() {
    }

    public Evenement(int idEv, String nomEv,String dateDEv, String dateFEv,String localisation) {
        this.idEv = idEv;
        this.nomEv = nomEv;
        this.dateDEv=dateDEv;
          this.dateFEv=dateFEv;
           this.localisation = localisation;
        
    }
public Evenement( String nomEv,String dateDEv, String dateFEv,String localisation) {
       
        this.nomEv = nomEv;
        this.dateDEv=dateDEv;
          this.dateFEv=dateFEv;
           this.localisation = localisation;
        
    }
    public int getIdEv() {
        return idEv;
    }

    public void setIdEv(int idEv) {
        this.idEv = idEv;
    }

    public String getNomEv() {
        return nomEv;
    }

    public void setNomEv(String nomEv) {
        this.nomEv = nomEv;
    }

    public String getLocalisation() {
        return localisation;
    }

    public void setLocalisation(String localisation) {
        this.localisation = localisation;
    }

    public String getDateDEv() {
        return dateDEv;
    }

    public String getDateFEv() {
        return dateFEv;
    }

    public void setDateDEv(String dateDEv) {
        this.dateDEv = dateDEv;
    }

    public void setDateFEv(String dateFEv) {
        this.dateFEv = dateFEv;
    }

    @Override
    public int hashCode() {
        int hash = 7;
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Evenement other = (Evenement) obj;
        if (this.idEv != other.idEv) {
            return false;
        }
        if (!Objects.equals(this.nomEv, other.nomEv)) {
            return false;
        }
        if (!Objects.equals(this.localisation, other.localisation)) {
            return false;
        }
        if (!Objects.equals(this.dateDEv, other.dateDEv)) {
            return false;
        }
        if (!Objects.equals(this.dateFEv, other.dateFEv)) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "Evenement{" + "idEv=" + idEv + ", nomEv=" + nomEv + ", localisation=" + localisation + ", dateDEv=" + dateDEv + ", dateFEv=" + dateFEv + '}';
    }

   
    

}

File: src/Entity/Evenement.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Serializer\Annotation\Groups;
/**
 * Evenement
 *
 * @ORM\Table(name="evenement")
 * @ORM\Entity
 *  @ORM\Entity(repositoryClass="App\Repository\EvenementRepository")
 */
class Evenement
{
    /**
     * @var int
     *
     * @ORM\Column(name="idEv", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    #[Groups("Evenement")]
    private $idev;

    /**
     * @var string
     *
     * @ORM\Column(name="nomev", type="string", length=255, nullable=false)
     */

     #[Assert\Length(min:5)]
    #[Assert\Length(max:20)]
    #[Assert\NotBlank (message:"veuillez saisir le nom de l'evenement ")]
    
    #[Groups("Evenement")]
    private $nomev;

  
    /**
     * @var \DateTimeInterface
     *
     * @ORM\Column(name="dateDEv", type="string", nullable=false)
     
     * @Assert\NotBlank(message="La date de début de l'événement est obligatoire  format must be YYYY-MM-DD.")
     * @Assert\LessThanOrEqual(propertyPath="datefev", message="La date de début de l'événement doit être antérieure ou égale à la date de fin.")
      *  @Assert\Regex(
 *     pattern="/^\d{4}-\d{2}-\d{2}$/",
 *     message="The date format must be YYYY-MM-DD"
 * )
     */
    
     #[Groups("Evenement")]
    private $datedev;

    /**
     * @var \DateTimeInterface
     *
     * @ORM\Column(name="dateFEv", type="string", nullable=false)
     * 
     * @Assert\NotBlank(message="La date de fin de l'événement est obligatoire format must be YYYY-MM-DD.")
     *  @Assert\Regex(
 *     pattern="/^\d{4}-\d{2}-\d{2}$/",
 *    
 * )
     */
    
     #[Groups("Evenement")]
    private $datefev;
     /**
     * @var string
     *
     * @ORM\Column(name="localisation", type="string", length=255, nullable=false)
     */
    #[Assert\Length(min:5)]
    #[Assert\Length(max:20)]
    #[Assert\NotBlank (message:"veuillez saisir le localisation  de l'evenement ")]
    
    #[Groups("Evenement")]
    private $localisation;

    /**
     * @var string
     *
     * @ORM\Column(name="ImageEv", type="string", length=255, nullable=false)
     */
    #[Assert\NotBlank (message:"veuillez saisir image  de l'evenement ")]
    
    #[Groups("Evenement")]
    private $imageev;

    public function getIdev(): ?int
    {
        return $this->idev;
    }
    public function setIdev(string $Idev): self
    {
  
          $this->idev = $Idev;
  
          return $this;
      }

    public function getNomev(): ?string
    {
        return $this->nomev;
    }

    public function setNomev(string $nomev): self
  {

        $this->nomev = $nomev;

        return $this;
    }

    public function getDatedev(): ?string
    {
        return $this->datedev;
    }

    public function setDatedev(string $datedev): self
    {
        $this->datedev = $datedev;

        return $this;
    }

    public function getDatefev(): ?string
    {
        return $this->datefev;
    }

    public function setDatefev(string $datefev): self
    {

       
        $this->datefev = $datefev;

        return $this;
    }

    public function getLocalisation(): ?string
    {
        return $this->localisation;
    }

    public function setLocalisation(string $localisation): self
    {
        $this->localisation = $localisation;

        return $this;
    }

    public function getImageev()
    {
        return $this->imageev;
    }

    public function setImageev($imageev)
    {
        $this->imageev = $imageev;

        return $this;
    }
    public function __toString(): string
    {
        return $this->nomev;
    }
   
    
    

}